codemark/firebase/database.py

```diff
@@ -81,28 +81,34 @@ def signout(self):
         except Exception as e:
             print_error(e)
 
-    def register(self, email, password):
-        '''
+    def register(self, email, password, user_type, user_config, user_type_config):
+        """
         Registers a user to the firebase realtime database 
         :param email: Email
         :param password: Password
-        '''
+        :param user_type: Whether the user is a 'student' or a 'professor'
+        :param user_config: Dictionary containing the details of the user
+            Keys: 'email', 'user_type', 'name', 'college', 'student_id'
+        :param user_type_config: Dictionary containing the details specific to a user_type
+            Keys (student): 'batch_id', 'batch_semester', 'college_name', 'email', 'field_of_study', 'name', 'passing_year', 'student_id'
+            Keys (professor): 'college_name', 'name', 'professor_id'
+        """
         try:
             user = pauth.create_user_with_email_and_password(email, password)
+            res = self.pushData('users', user_config)
+            res = self.pushData(user_type, user_type_config)
```

Comment:
> res = self.pushData(user_type, user_type_config)

you should perform appropriate checks on user_type here, so that it is not abused, 

also in my opinion get them segregated from this function
push to users is completely ok, but one to professor is kinda edgy, and will sooner or later may break the APIs in future.

if you wanna discuss possible design beforehand, you're more than welcome